src/secrets/KeyVaultProvider.ts

```
@@ -45,17 +47,14 @@ export class KeyVaultProvider {
         }
     }
 
-    /**
-     * Initialized the KeyVault client.
-     * This is handled in a separate method to avoid calling async operations in the constructor.
-     */
-    private async _initialize() {
+    private async initialize() {
 
         // Use specified authentication type (either MSI or CLI)
         const creds: any = this.authType === "MSI" ?
             await msRestNodeAuth.loginWithAppServiceMSI({ resource: "https://vault.azure.net" }) :
             await msRestNodeAuth.AzureCliCredentials.create({ resource: "https://vault.azure.net" });
 
         this.client = new SecretClient(this.url, creds);
+        return;
```

the client isn't fully initialized until you do a round trip to the key vault. With the fail early approach, you should test that here, throw a friendly error and exit(-1)

This is also a good place to put the retry logic necessary for pod identity.

---

This is related to the conversation we had about where/when a failed credential authentication error would appear with this object ready pattern.  Maybe also/instead in config.ts when we await keyvault.ready -- if there is a way to catch/react to an authentication failure.

---

So I think we just add a `try` in the initializer so we can _explicitly_ log a friendly error, right?

---

I just pushed a commit adding a `try` around the keyvault initialize function.

---

There's an issue for making this work with pod identity, so we can address the full round-trip then.